From 11801886c5e3fe0fd1d6dbb353412930c3930121 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:49:12 -0400
Subject: [tools] perf/trace: Honour -m option

Message-id: <1408953210-26343-65-git-send-email-jolsa@redhat.com>
Patchwork-id: 88788
O-Subject: [PATCH RHEL7.1 BZ1133083 064/322] perf trace: Honour -m option
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit f885037ef6400ce4a4c122a88845dea2c9bca256
Author: Jiri Olsa <jolsa@redhat.com>
Date: Thu Nov 28 17:57:22 2013 +0100

description
===========
Currently trace command supports '-m' option, but does not honours its
value and keeps the default.

Changing the perf_evlist__mmap function call to use the '-m' configured
value.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index e9f345e..9f2a242 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -1890,7 +1890,7 @@ static int trace__run(struct trace *trace, int argc, const char **argv)
  if (err < 0)
   goto out_error_open;
 
- err = perf_evlist__mmap(evlist, UINT_MAX, false);
+ err = perf_evlist__mmap(evlist, trace->opts.mmap_pages, false);
  if (err < 0) {
   fprintf(trace->output, "Couldn't mmap the events: %s\n", strerror(errno));
   goto out_close_evlist;
-- 
1.7.1